test/extended/cmd.sh

```diff
@@ -54,88 +54,89 @@ oadm policy add-role-to-group view system:authenticated -n default
-os::cmd::expect_success "docker tag -f docker.io/busybox:latest ${DOCKER_REGISTRY}/image-ns/busybox:latest"
-os::cmd::expect_success "docker push ${DOCKER_REGISTRY}/image-ns/busybox:latest"
-os::cmd::expect_success "docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest"
-
-
-DOCKER_CONFIG_JSON=${HOME}/.docker/config.json
-oc new-project dc-ns
-os::cmd::expect_success "oc delete all --all"
-os::cmd::expect_success "oc delete secrets --all"
-os::cmd::expect_success "oc secrets new image-ns-pull .dockerconfigjson=${DOCKER_CONFIG_JSON}"
-os::cmd::expect_success "oc secrets new-dockercfg image-ns-pull-old --docker-email=fake@example.org --docker-username=imagensbuilder --docker-server=${DOCKER_REGISTRY} --docker-password=${IMAGE_NS_TOKEN}"
-
-os::cmd::expect_success "oc process -f test/extended/fixtures/image-pull-secrets/pod-with-no-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - "
-os::cmd::try_until_text "oc describe pod/no-pull-pod" 'Back-off pulling image'
-os::cmd::expect_success "oc delete pods --all"
-
-os::cmd::expect_success "oc process -f test/extended/fixtures/image-pull-secrets/pod-with-new-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - "
-os::cmd::try_until_text 'oc get pods/new-pull-pod -o jsonpath={.status.containerStatuses[0].imageID}' 'docker'
-os::cmd::expect_success "oc delete pods --all"
-os::cmd::expect_success "docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest"
-
-os::cmd::expect_success "oc process -f test/extended/fixtures/image-pull-secrets/pod-with-old-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - "
-os::cmd::try_until_text 'oc get pods/old-pull-pod -o jsonpath={.status.containerStatuses[0].imageID}' 'docker'
-os::cmd::expect_success "oc delete pods --all"
-os::cmd::expect_success "docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest"
-
-os::cmd::expect_success "oc process -f test/extended/fixtures/image-pull-secrets/dc-with-old-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - "
-os::cmd::try_until_text 'oc get pods/my-dc-old-1-hook-pre -o jsonpath={.status.containerStatuses[0].imageID}' 'docker'
-os::cmd::expect_success "oc delete all --all"
-os::cmd::expect_success "docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest"
-
-os::cmd::expect_success "oc process -f test/extended/fixtures/image-pull-secrets/dc-with-new-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - "
-os::cmd::try_until_text 'oc get pods/my-dc-1-hook-pre -o jsonpath={.status.containerStatuses[0].imageID}' 'docker'
-os::cmd::expect_success "oc delete all --all"
-os::cmd::expect_success "docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest"
+os::cmd::expect_success 'oc new-project image-ns'
+os::cmd::expect_success 'oc delete all --all'
+os::cmd::expect_success "docker login -u imagensbuilder -p '$( oc sa get-token builder )' -e fake@example.org '${docker_registry}'"
```

Comment:
I was under the impression that this would be expanded inside of `os::cmd` and therefore keep the 'title' of our test from containing the (very long and runtime-unique) secret string, which would help formatting and history tracking. It seems as though it's expanded early, so that may not be the case. I'll look into it.